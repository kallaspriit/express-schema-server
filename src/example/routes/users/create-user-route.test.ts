import * as supertest from 'supertest';
import setupApp from '../../app';

let app: supertest.SuperTest<supertest.Test>;

describe('create-user-route', () => {
	beforeEach(async () => {
		app = supertest(await setupApp());
	});

	it('should register valid user', async () => {
		const response = await app.post('/users').send({
			name: 'Jack Daniels',
			email: 'jack@daniels.com',
		});

		expect(response.status).toEqual(200);
		expect(response.body.success).toBe(true);
		expect(response.body.payload).toMatchSnapshot();
	});

	it('should return validation error for invalid email', async () => {
		const response = await app.post('/users').send({
			name: 'Jack Daniels',
			email: 'jack@daniels',
		});

		expect(response.status).toEqual(400);
		expect(response.body.success).toBe(false);
		expect(response.body.payload).toMatchSnapshot();
	});

	it('should return validation error for duplicate email', async () => {
		const response1 = await app.post('/users').send({
			name: 'Jack Daniels',
			email: 'jack@daniels.com',
		});

		expect(response1.status).toEqual(200);
		expect(response1.body.success).toBe(true);
		expect(response1.body.payload).toMatchSnapshot();

		const response2 = await app.post('/users').send({
			name: 'Jack Daniels',
			email: 'jack@daniels.com',
		});

		expect(response2.status).toEqual(400);
		expect(response2.body.success).toBe(false);
		expect(response2.body.payload).toMatchSnapshot();
	});
});
