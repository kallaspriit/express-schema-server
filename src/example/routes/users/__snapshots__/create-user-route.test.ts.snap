// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create-user-route should register valid user 1`] = `
Object {
  "email": "jack@daniels.com",
  "id": 1,
  "name": "Jack Daniels",
}
`;

exports[`create-user-route should return validation error for duplicate email 1`] = `
Object {
  "email": "jack@daniels.com",
  "id": 1,
  "name": "Jack Daniels",
}
`;

exports[`create-user-route should return validation error for duplicate email 2`] = `null`;

exports[`create-user-route should return validation error for invalid email 1`] = `null`;
